src/main.js

```diff
@@ -1,84 +1,73 @@
-// Private Methods
-function extendConfig(defaults, options) {
-    let prop, extended = {};
-
-    for (prop in defaults) {
-        if (Object.prototype.hasOwnProperty.call(defaults, prop)) {
-            extended[prop] = defaults[prop];
-        }
-    }
-
-    for (prop in options) {
-        if (Object.prototype.hasOwnProperty.call(defaults, prop)) {
-            extended[prop] = options[prop];
-        }
-    }
-
-    return extended;
-};
-
 export default class Concertina {
-    constructor(el, options) {
+    constructor(el, customOptions) {
         // Config
         this.element = el; // TODO: Get this to work with multiple elements - document.querySelectorAll('blah');
-        this.defaults = {
+        this.options = {
             'blockClass': '.Concertina',
             'tabClass': '.Concertina-tab',
             'tabActiveClass': '.is-selected',
             'panelClass': '.Concertina-panel',
             'panelActiveClass': '.is-visible'
         };
-        this.options = extendConfig.bind(this)(this.defaults, options);
-        this.tabs = this.element.querySelectorAll(this.options.tabClass);
-        this.panels = this.element.querySelectorAll(this.options.panelClass);
+
+        for (let prop in customOptions) {
+            this.options[prop] = customOptions[prop];
+        }
+
+        this.tabs = [...this.element.querySelectorAll(this.options.tabClass)];
+        this.panels = [...this.element.querySelectorAll(this.options.panelClass)];
     }
 
     init() {
         //TODO: Set aria-controls attr on tabs and aria-labelledby attr on panels.
+
         this.setTabsInactive();
         this.setPanelsInactive();
 
         this.element.addEventListener('click', (e) => {
-            if (e.target.classList.contains(this.options.tabClass.replace('.', ''))){
+            if (this.tabs.includes(e.target)){
```

Comment:
Because the NodeList is now an Array we can use fancy new ES6 methods to make our code cleaner and easier to reason about. (we don't have to use the class string as an intermediary anymore: just compare the objects directly!)